### PR TITLE
issues on friday

### PR DESCRIPTION
**Pull request by srikanthgudipudi**

### CODE REVIEW

File: src/app/modules/admin-shared/components/multi-select/multi-select.component.ts [hunk 1]
```diff
@@ -1,9 +1,14 @@
-import { Component, OnInit } from '@angular/core';
-import { FormGroup } from "@angular/forms";
+// import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormControl, FormControlName } from "@angular/forms";
 import { FieldConfig } from "../../field.interface";
 import { Observable } from 'rxjs';
 import { map, startWith } from 'rxjs/operators';
 
+import { COMMA, ENTER } from '@angular/cdk/keycodes';
```
Comment (kiranharidas187) [inline src/app/modules/admin-shared/components/multi-select/multi-select.component.ts]: please remove unwanted imports

File: src/app/modules/admin-shared/components/multi-select/multi-select.component.ts [hunk 2]
```diff
@@ -17,14 +22,32 @@ export class MultiSelectComponent implements OnInit {
...
+  selectable = true;
+  removable = true;
+  addOnBlur = false;
+  separatorKeysCodes: number[] = [ENTER, COMMA];
+  // fruitCtrl = new FormControl();
+  filtereddata: any;
+  fruits: any[] = [];
+
+  @ViewChild('searchInput') searchInput: ElementRef<HTMLInputElement>;
+  @ViewChild('searchInput', { read: MatAutocompleteTrigger }) autoComplete;
```
Comment (kiranharidas187) [inline src/app/modules/admin-shared/components/multi-select/multi-select.component.ts]: Why this 2 ref?

File: src/app/modules/admin-shared/components/multi-select/multi-select.component.ts [hunk 3]
```diff
@@ -72,6 +95,78 @@ export class MultiSelectComponent implements OnInit {
...
+
+  remove(fruit){
+    console.log('remove', fruit);
+    const index = this.fruits.indexOf(fruit);
+    if (index >= 0) {
+      this.fruits.splice(index, 1);
+    }
+  }
+
+  onFruitSelectionChange(event: MatAutocompleteSelectedEvent): void {
```
Comment (kiranharidas187) [inline src/app/modules/admin-shared/components/multi-select/multi-select.component.ts]: give valid function names

File: src/app/modules/admin-shared/components/multi-select/multi-select.component.ts [hunk 4]
```diff
@@ -72,6 +95,78 @@ export class MultiSelectComponent implements OnInit {
...
+    console.log('uuuuuuuuuu', this.fruits);
+  }
+
+  isFruitSelected(fruit): boolean {
+    // console.log('isFruitSelected', fruit);
+    return this.fruits.indexOf(fruit) >= 0;
+  }
+
+  updateFruitList(fruit) {
+    console.log('updateFruitList', fruit);
```
Comment (kiranharidas187) [inline src/app/modules/admin-shared/components/multi-select/multi-select.component.ts]: remove all unwanted console logs

File: src/app/modules/users/users-list/users-list.component.ts [hunk 5]
```diff
@@ -4,13 +4,13 @@ import { MatDialog, MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dial
...
-import { fromEvent} from 'rxjs';
+import { fromEvent } from 'rxjs';
 import { AddMultipleUsersComponent } from '../add-multiple-users/add-multiple-users.component';
 import { UsersService } from '../../admin-core';
 import { SelectionModel } from '@angular/cdk/collections';
-import { distinctUntilChanged, map,filter } from 'rxjs/operators';
+import { distinctUntilChanged, map, filter } from 'rxjs/operators';
 import { MatSnackBar } from '@angular/material';
-import {saveAs as importedSaveAs} from "file-saver";
+import { saveAs as importedSaveAs } from "file-saver";
```
Comment (kiranharidas187) [inline src/app/modules/users/users-list/users-list.component.ts:14]: remove all unwanted imports

File: src/app/modules/admin-shared/components/multi-select/multi-select.component.ts [hunk 6]
```diff
@@ -14,64 +16,26 @@ export class MultiSelectComponent implements OnInit {
...
-    }
-    return displayValue;
   }
 
-  // To select the data 
-  optionClicked(event: Event, user) {
-    event.stopPropagation();
-    this.toggleSelection(user);
-  }
+  protected _onDestroy = new Subject<void>();
```
Comment (kiranharidas187) [inline src/app/modules/admin-shared/components/multi-select/multi-select.component.ts]: Why u need subject?

File: src/app/modules/admin-shared/components/multi-select/multi-select.component.ts [hunk 7]
```diff
@@ -1,8 +1,10 @@
...
+import { FormGroup, FormControl, FormControlName } from "@angular/forms";
 import { FieldConfig } from "../../field.interface";
-import { Observable } from 'rxjs';
-import { map, startWith } from 'rxjs/operators';
+import { ReplaySubject, Subject } from 'rxjs';
+import { Component, ElementRef, ViewChild, OnInit, Input, QueryList, ViewChildren } from '@angular/core';
+import { MatAutocompleteSelectedEvent, MatAutocompleteTrigger, MatChipInputEvent } from '@angular/material';
+import { take, takeUntil } from 'rxjs/operators';
 
 @Component({
```
Comment (kiranharidas187) [inline src/app/modules/admin-shared/components/multi-select/multi-select.component.ts:7]: Remove all unwanted imports

File: src/app/modules/admin-shared/components/multi-select/multi-select.component.ts (hunk 1 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-shared/components/multi-select/multi-select.component.ts]: Removed

File: src/app/modules/admin-shared/components/multi-select/multi-select.component.ts (hunk 2 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-shared/components/multi-select/multi-select.component.ts]: Removed

File: src/app/modules/admin-shared/components/multi-select/multi-select.component.ts (hunk 3 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-shared/components/multi-select/multi-select.component.ts]: Removed

File: src/app/modules/admin-shared/components/multi-select/multi-select.component.ts (hunk 4 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-shared/components/multi-select/multi-select.component.ts]: Removed

File: src/app/modules/users/users-list/users-list.component.ts (hunk 5 above)
Comment (srikanthgudipudi) [reply src/app/modules/users/users-list/users-list.component.ts:14]: it is required bro

File: src/app/modules/admin-shared/components/multi-select/multi-select.component.ts (hunk 6 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-shared/components/multi-select/multi-select.component.ts]: Removed

File: src/app/modules/admin-shared/components/multi-select/multi-select.component.ts (hunk 7 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-shared/components/multi-select/multi-select.component.ts:7]: Removed

**Pull request merged**